### PR TITLE
Fix handlebars path expressions starts with `@`

### LINKED ISSUE #16356: handlebars attrbutes changed: {{@x.y.z}} becomes {{@x}}
**Prettier 3.3.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEACVAeAJgSwG7qFgA2AhgM7kC8AOiAjAE4CedhqAfDVOhgBYBGDsGDoYOGMTjsAvjIwB6QVx6ZcBdCQrU6AIwhZWIFexEABAB4A6ZlYBec7u2eERhfYZeFHUReo6YCv4gADQgEAAO4tDkyKCkjIwQAO4ACgkIsSikxMmkzLFhuoykYADWcDAAyqQAtnAAMjhQcMgAZjnkcEUl5ZVVEaXNAObITACu3SBdtThjjJNhcBYRcIw49bA5ACprUAk4cFkdxF1h5CNSAIrjEPDtnVMAVuQWVZdwN3etSCdnIABHW7wVJJCJZEAUAC0LTgWDhoRATFIOGIIwAwhBarVSMhIcRiIiLlBhlIAIIwJg4XTjEFrJotB6nKZ8GC1YgAdT4EiOgzAcCqmQk+AkzDxYEoiLwkwAklB4bAqmB1lEyfKqjBmFImf8IkkuhyShE8Xqjms8K0ws0uowYKDSMMcTqpoNGDa8aSNvVGIi9c0YBycFgYHxkAAOAAMYUYcCBOBj9sduN+jzCMFIukDwdDSAATGFxl1thnjqn6LVdHD4VgGqQSeMHXAAGIQRg4ykjPGkWkQEByIA)

```sh
--parser glimmer
```

**Input:**

```hbs
  <div     class="entry"    >
  <h1>{{  title    }}</h1>
  <div   class="body">
    {{@x.y.z}}
            {{   body         }}
</div> </div>
```

**Output:**

```hbs
<div class="entry">
  <h1>{{title}}</h1>
  <div class="body">
    {{@x}}
    {{body}}
  </div>
</div>
```

**Expected behavior:**
Starting with 3.3.1 {{@x.y.z}} is output as {{@x}} , if I add `--debug-check` it will catch that the output is different

### PR DESCRIPTION
**Pull request by Princeyadav05**
## Description

Fixed #16356
Changed print('head') to print(node.tail) as mentioned in the comment: https://github.com/prettier/prettier/issues/16356#issuecomment-2150771806

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-handlebars/printer-glimmer.js [hunk 1]
```diff
@@ -793,7 +793,7 @@ const isPathExpressionPartNeedBrackets = (part, index) =>
   );
 function printPathExpression(node, print) {
   if (node.head.type === "AtHead") {
-    return print("head");
+    return print(node.tail);
```
Comment (fisker) [inline src/language-handlebars/printer-glimmer.js:796]:
```suggestion
    return join(".", [print("head"), ...node.tail]);
```

You should also add tests to confirm it work as expected.  Take https://github.com/prettier/prettier/pull/15605/files as example.

Comment (Princeyadav05) [reply src/language-handlebars/printer-glimmer.js]: Will have to look for tests.

Comment (fisker) [reply src/language-handlebars/printer-glimmer.js]: Here is the guide for tests https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md#tests

Also, don't forgot add a changelog entry.

Comment (Princeyadav05) [reply src/language-handlebars/printer-glimmer.js]: @fisker has tried to add a test case and followed the guides. However, this is my first contribution towards a test case. Please have a look and guide me on if anything needs to be changed.

Comment (fisker) [reply src/language-handlebars/printer-glimmer.js]: Good job, I can do the rest of the work.

Comment (fisker) [reply src/language-handlebars/printer-glimmer.js]: Turns out it seems more then add `node.tail`, we have to use the existing `isPathExpressionPartNeedBrackets` logic. We are good now.

**Pull request merged**